tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,134 @@
+      payload.toString('utf8fix/7136 ').split('\n').forEach(line => {
+        const [sha, platform] = line.split('  ');
+        shaSums[platform] = sha;
       });
+      grunt.log.ok(`sha sums download complete`);
     });
+  };
 
-    // use an async iife to store promise for download
-    // then store platform in active downloads list
-    // which we will read from in the finish task
-    platform.downloadPromise = (async function () {
-      grunt.file.mkdir(downloadDir);
+  const checkShaSum = (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums[basename(dirname(platform.nodeUrl)) + '/' + file] : shaSums[file]
+    };
+
+    return checkHashFromFile(path, expected).then(([passed, actual]) => {
+      if (passed) grunt.log.ok(`${platform.name} shasum verified`);
+      else grunt.log.error(`${platform.name} shasum check failed`);
+      return passed;
+    });
+  };
 
-      if (platform.win) {
-        await fromNode(cb => {
-          resp
-          .pipe(createWriteStream(resolve(downloadDir, 'node.exe')))
-          .on('error', cb)
-          .on('finish', cb);
-        });
-      } else {
-        await fromNode(cb => {
-          resp
-          .pipe(createGunzip())
-          .on('error', cb)
-          .pipe(new Extract({ path: downloadDir, strip: 1 }))
-          .on('error', cb)
-          .on('end', cb);
-        });
+  const getNodeBuild = (platform) => {
+    const finalDir = platform.nodeDir;
+    const downloadDir = `${finalDir}.temp`;
+    const file = basename(platform.nodeUrl);
+
+    return wreckGet(platform.nodeUrl)
+    .then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`);
       }
+      const bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
+      const mb = ((bytes / 1024) / 1024).toFixed(2);
+      grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
+      grunt.file.mkdir(downloadDir);
+      return payload;
+    })
+    .then(payload => {
+      return writeFileAsync(resolve(downloadDir, file), payload);
+    })
+    .then(() => {
+      return renameAsync(downloadDir, finalDir);
+    });
 
-      await fromNode(cb => {
-        rename(downloadDir, finalDir, cb);
+  };
+
+  const extractNodeBuild = async (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+
+    await fromNode(cb => {
+      createReadStream(path)
+      .pipe(createGunzip())
+      .on('error', (err) => {
+        cb(new Error(`error extracting: ${err.message}`));
+      })
+      .pipe(new Extract({path: platform.nodeDir, strip: 1}))
+      .on('error', (err) => {
+        cb(new Error(`error extracting: ${err.message}`));
+      })
+      .on('end', () => {
+        cb(null, true);
       });
```

Comment:
I think the original code was misusing `fromNode` and we can clean this up a little by simply creating a new promise:

```
    return new Promise((resolve, reject) => {
      createReadStream(path)
      .pipe(createGunzip())
      .on('error', reject)
      .pipe(new Extract({path: platform.nodeDir, strip: 1}))
      .on('error', reject)
      .on('end', resolve);
    });
```